en/docs/visual-programming/data-mapper.md

```diff
@@ -1 +1,160 @@
+
+### Basic Mapping
+Map the `person id` to the `student id` as shown below.
+>**Info:** The mapping view will have the **Inputs** on the left hand side of the UI and the **Output** on the right. To map the fields, click on the input field port and then, click the output field port. If the input and output fields are compatible and can be mapped directly, you will see a solid line connecting them.
+
+![Basic Mapping](img/dm-basic-mapping.gif "Basic mapping")
+
+### Diagnose and fix mapping errors
+Use the `toBalString` lang lib function to convert the int to string as shown below.
+
+>**Info:** When you map the input to an output fields, some of them might not be compatible due to type mismatch. In this example, if you map the `person age` to `student` age, it will result in an error for type mismatch since the `input age` type is an integer and `output age` type is string. In this case, the Data Mapper will connect the two fields with a red line and show an alert sign. You can see the error by hovering over the alert sign. In this case, it will show `incompatible types: expected 'string', found 'int'`. To fix the error, hover over the alert sign and click `Fix by editing expression`. Then, the Data Mapper will pop out the expression editor for the specific expression. Now, you can modify the expression to return a string.
+
+![Fix Errors](img/dm-fix-diagnostics.gif "Fix incompatible types error")
+
+Once you fix the error, the connection appears in blue to indicate that there are no errors.
+
+### Aggregate multiple input fields to one output field
+You can derive one output parameter by combining two or more input parameters. In this example, the value for the fullName output parameter is a combination the values for the firstName and lastName input parameters. You can map them as shown below:
+>**Info:** To aggregate fields, you can map two or more fields to the same output field. The Data Mapper will automatically combine the two fields and assign it to the output field. By default, the fields will be combined with a plus operator. If you want to use a different operator or method to combine two fields, you can click on the code button and customize the expression with the expression editor.
+
+![Concatenate](img/dm-concatination.gif "Aggregate multiple input fields")
+
+### Map the arrays
+To convert from one array type to another, you can simply map the input array to the output array. If the arrays are compatible, they will be connected with a blue line. If they are not compatible, the connecting line will appear in red.
+
+>**Info:** You can use Ballerina query support to convert one array type to another. To use a query in a Data Mapper, you can select the array by clicking on it. Then, it will provide you with several buttons. Click the code action button (bulb sign) and select **Convert to query**. Then, the Data Mapper will convert the mapping to a query. Then, move into the query and do the mapping between the array types.
```

Comment:
```suggestion
>**Info:** You can use the Ballerina query support to convert one array type to another. To use a query in the Data Mapper, you can select the array by clicking on it. Then, it will provide you with several buttons. Click the code action button (bulb sign) and select **Convert to query**. Then, the Data Mapper will convert the mapping to a query. Then, do the mapping between the query and the array types.
```

Reply:
Here we should mention go inside the query and do the mappings